docs/howto/operator/python.rst

```diff
@@ -34,7 +34,8 @@ Passing in arguments
 ^^^^^^^^^^^^^^^^^^^^
 
 Use the ``op_args`` and ``op_kwargs`` arguments to pass additional arguments
-to the Python callable.
+to the Python callable. If you use any of the :doc:`context variables <../../macros-ref>`
+as an argument of the provided callable, the value will be automatically injected as shown below:
```

Comment:
> All context variables can still be provided with a double-asterisk argument:
> 
> ```python
> def myfunc(**context):
>     print(context)  # all variables will be provided to context
> 
> ```
> 
> 
> python_operator = PythonOperator(task_id='mytask', python_callable=myfunc)

I think this part of UPDATING.md is missing in docs. 

In my opinion, there is still too much documentation in the UPDATING.md file.

Maybe it's worth moving some content from this file and linking to other documentation? I mentioned this file, but docstring is also useful documentation. This can discuss reserved parameter names ..

Reply:
I think the main goal was to simplify how it works, that you don't have to care about how this works as a user of Airflow. Besides that, I don't like the `**` notation:
```python
def myfunc(**context):
    print(context['ds'])
```
is equivalent to:
```python
def myfunc(ds):
    print(ds)
```
So I don't want to point the user in the direction of using an overly verbose way of doing this. Normally in Python you should also be conscious with the kwargs stuff.

Reply:
There aren't really reserved parameter names anymore. The only clash that is possible if you do something like:
```
def fn(dag, ds, **context):
    print(dag)

PythonOperator(
    op_args=[1],
    python_callable=fn
)
```
In this case the `dag == 1`, and the `ds` is just the execution date. If you would change the code to:
```
def fn(dag, ds, **context):
    print(dag)

PythonOperator(
    op_args=[1, 2],
    python_callable=fn
)
```
Then `ds` would be `ds == 2`. To avoid confusion the arguments that you provide using the `op_args` cannot be part of the keywords in the context. This is an edge case that will not happen in practice, and we throw an error just to keep the users their sanity.